### PR TITLE
Split chat.run in sub functions

### PR DESCRIPTION
**Pull request by GuillemineA**
chat.run remains the function to run a chat instance
it call either "buildOpenAIPayload" or "buildGeminiPayload"
it calls callGenAiAPI
it calls either "handleOpenAIToolCalls" or "handleGeminiToolCalls"

## Summary by CodeRabbit

- **New Features**
	- Enhanced chat functionalities to support both OpenAI and Gemini APIs.
	- Introduced new methods for building payloads tailored for the Gemini API.
	- Added refined handling for tool calls from the Gemini API, improving response management.
	- Updated methods to accommodate new payload structures for better interaction with both APIs.
	- Updated README to reflect new API key requirements and usage instructions for Gemini integration.

- **Bug Fixes**
	- Improved error handling and payload construction for API interactions.

### CODE REVIEW

File: src/code.gs [hunk 1]
```diff
@@ -570,61 +663,95 @@ const GenAIApp = (function () {
...
-                let responseMessage;
-                if (numberOfAPICalls <= maximumAPICalls) {
-                    let endpointUrl = "https://api.openai.com/v1/chat/completions";
-                    if (model.includes("gemini")) {
-                        payload = convertPayloadForGemini(payload);
-                        endpointUrl = `https://${region}-aiplatform.googleapis.com/v1/projects/${gcpProjectId}/locations/${region}/publishers/google/models/${model}:generateContent`;
+                return payload;
+            }
 
+            this.buildGeminiPayload = function (advancedParametersObject) {
```
Comment (keligggg) [inline src/code.gs:669]: Ya des fonctions comme ça qui sont dans l'auto completion de app script, je trouve ça pas fou
Si jamais tu trouves un moyen de les retirier

**Comment by keligggg** [comment]
Dans le message d'erreur "Please set your GCP project auth using GenAIApp.setGeminiAuth({project-id: 'YOUR_PROJECT_ID', region: 'REGION'})""

Ajoute des guillemets pour sur les clés pour aider au copier coller "Please set your GCP project auth using GenAIApp.setGeminiAuth({'project-id': 'YOUR_PROJECT_ID', 'region': 'REGION'})"

**Comment by keligggg** [comment]
Dans "
    setGeminiAuth: function (authObject) {
      gcpProjectId = authObject.project_id;
      region = authObject.project_id
    },"

pourquoi ne pas mettre project_id et project_id en paramètre avec uen doc ça aide

**Comment by GuillemineA** [comment]
@coderabbitai full review

File: src/code.gs [hunk 2]
```diff
@@ -397,8 +406,8 @@ const GenAIApp = (function () {
                     if (advancedParametersObject.model) {
```
Comment (keligggg) [inline src/code.gs:406]: Parfois tu marques youAPIKey --> yourAPIKey

File: src/code.gs [hunk 3]
```diff
@@ -704,106 +881,166 @@ const GenAIApp = (function () {
         }
```
[the remark this answers is not in the document]
Comment (keligggg) [reply src/code.gs:881]: Et les 403 quand un mauvais projet GCP est choisi

File: src/code.gs [hunk 4]
```diff
@@ -231,6 +239,7 @@ const GenAIApp = (function () {
             };
 
             /**
+             * NOT FOR GEMINI
```
Comment (keligggg) [inline src/code.gs:242]: Only for Open ai ?

**Pull request merged**